src/components/dialogs/network-modifications/battery/creation/battery-creation-dialog.js

```diff
@@ -0,0 +1,257 @@
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { yupResolver } from '@hookform/resolvers/yup';
+import yup from 'components/utils/yup-config';
+import {
+    EQUIPMENT_ID,
+    EQUIPMENT_NAME,
+    CONNECTION_NAME,
+    CONNECTIVITY,
+    ID,
+    VOLTAGE_LEVEL,
+    BUS_OR_BUSBAR_SECTION,
+    CONNECTION_DIRECTION,
+    CONNECTION_POSITION,
+    ACTIVE_POWER_SET_POINT,
+    REACTIVE_POWER_SET_POINT,
+    ACTIVE_POWER,
+    REACTIVE_POWER,
+    MAXIMUM_REACTIVE_POWER,
+    MINIMUM_REACTIVE_POWER,
+    MAXIMUM_ACTIVE_POWER,
+    MINIMUM_ACTIVE_POWER,
+    FREQUENCY_REGULATION,
+    DROOP,
+    REACTIVE_CAPABILITY_CURVE_TABLE,
+    REACTIVE_CAPABILITY_CURVE_CHOICE,
+} from 'components/utils/field-constants';
+import {
+    getConnectivityWithPositionEmptyFormData,
+    getConnectivityFormData,
+    getConnectivityWithPositionValidationSchema,
+} from '../../../connectivity/connectivity-form-utils';
+import BatteryCreationForm from './battery-creation-form';
+import { createBattery } from 'utils/rest-api';
+import { sanitizeString } from '../../../dialogUtils';
+import {
+    FORM_LOADING_DELAY,
+    UNDEFINED_CONNECTION_DIRECTION,
+} from 'components/network/constants';
+import {
+    getReactiveLimitsEmptyFormData,
+    getReactiveLimitsSchema,
+} from '../../generator/reactive-limits/reactive-limits-utils';
+import { useOpenShortWaitFetching } from 'components/dialogs/commons/handle-modification-form';
+import { FetchStatus } from 'utils/rest-api';
+import { getFrequencyRegulationSchema } from '../../generator/set-points/set-points-utils';
+
+const emptyFormData = {
+    [EQUIPMENT_ID]: '',
+    [EQUIPMENT_NAME]: '',
+    [MAXIMUM_ACTIVE_POWER]: null,
+    [MINIMUM_ACTIVE_POWER]: null,
+    [ACTIVE_POWER]: null,
+    [REACTIVE_POWER]: null,
+    [ACTIVE_POWER_SET_POINT]: null,
+    [REACTIVE_POWER_SET_POINT]: null,
+    [FREQUENCY_REGULATION]: false,
+    [DROOP]: null,
+    ...getReactiveLimitsEmptyFormData(),
+    ...getConnectivityWithPositionEmptyFormData(),
+};
+
+const formSchema = yup
+    .object()
+    .shape(
+        {
+            [EQUIPMENT_ID]: yup.string().required(),
+            [EQUIPMENT_NAME]: yup.string(),
+            [MAXIMUM_ACTIVE_POWER]: yup.number().nullable().required(),
+            [MINIMUM_ACTIVE_POWER]: yup.number().nullable().required(),
+            [ACTIVE_POWER]: yup.number().nullable().required(),
+            [REACTIVE_POWER]: yup.number().nullable().required(),
+            ...getReactiveLimitsSchema(),
+            ...getConnectivityWithPositionValidationSchema(),
+            ...getFrequencyRegulationSchema(),
+        },
+        [MAXIMUM_REACTIVE_POWER, MINIMUM_REACTIVE_POWER]
+    )
+    .required();
+
+const BatteryCreationDialog = ({
+    editData,
+    currentNode,
+    studyUuid,
+    isUpdate,
+    editDataFetchStatus,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode.id;
+    const { snackError } = useSnackMessage();
+
+    const equipmentPath = 'generators';
```

Comment:
it is better to make constants outside the component (performance impact due to the renders)
 const EQUIPMENT_PATH = 'generators';